action/protocol/execution/evm/evm.go

```diff
@@ -316,6 +300,27 @@ func ExecuteContract(
 	return retval, receipt, nil
 }
 
+func depositeGas(ctx context.Context, sm protocol.StateManager, execution *action.Execution, consumedGas uint64, gasPrice *big.Int, hCtx HelperContext, fCtx protocol.FeatureCtx) (*action.TransactionLog, error) {
+	var (
+		receiver                  address.Address
+		sharedGas                 uint64
+		sharedGasFee, totalGasFee *big.Int
+		err                       error
+	)
+	if fCtx.SharedGasWithDapp && hCtx.Sgd != nil {
+		receiver, sharedGas, err = processSGD(ctx, sm, execution, consumedGas, hCtx.Sgd)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to process Sharing of Gas-fee with DApps")
+		}
+	}
+	if sharedGas > 0 {
+		sharedGasFee = big.NewInt(int64(sharedGas))
+		sharedGasFee.Mul(sharedGasFee, gasPrice)
+	}
+	totalGasFee = new(big.Int).Mul(new(big.Int).SetUint64(consumedGas), gasPrice)
+	return hCtx.DepositGasFunc(ctx, sm, receiver, totalGasFee, sharedGasFee)
+}
```

Comment:
this is not what I mean.

```
func processSGD(ctx context.Context, sm protocol.StateManager, execution *action.Execution, consumedGas uint64, sgd SGDRegistry,
) (address.Address, uint64, error) {
    if sgd == nil {
        return nil, 0, nil
    }
    ....
```
thus, outside, we only need to check`if ps.featureCtx.SharedGasWithDapp`

Reply:
reverted, and add TODO